File: LeetCode/RotateArray.java
/**
 * Created by root on 2/26/15.
 */
public class RotateArray {
    public static void rotate(int[] nums, int k) {
        int N = nums.length;
        int[] temp = new int[N];
        int index = 0;
        k = k % N;
        for (int i = N - k; i < N; i++) {
            temp[index++] = nums[i];
        }
        for (int i = 0; i < N - k; i++) {
            temp[index++] = nums[i];
        }
        for (int i = 0; i < N; i++) {
            nums[i] = temp[i];
        }
    }
}

File: LeetCode/Python/RotateArray.py
__author__ = 'root'
class Solution:
    # @param nums, a list of integer
    # @param k, num of steps
    # @return nothing, please modify the nums list in-place.
    def rotate(self, nums, k):
        n = len(nums)
        k %= n
        temp = []
        for i in range(k)[::-1]:
            temp.append(nums[-i-1])
        for i in range(n - k):
            temp.append(nums[i])
        for i in range(n):
            nums[i] = temp[i]

a = [1,2]
ak = Solution
print(ak.rotate(ak, a, 1))